rcljava/src/test/java/org/ros2/rcljava/node/NodeParametersTest.java

```diff
@@ -0,0 +1,383 @@
+    List<ParameterVariant> plist = new ArrayList<ParameterVariant>();
+    List<rcl_interfaces.msg.ParameterDescriptor> pdlist = new ArrayList<rcl_interfaces.msg.ParameterDescriptor>();
+
+    plist.add(new ParameterVariant("foo", true));
+    pdlist.add(new rcl_interfaces.msg.ParameterDescriptor());
+
+    node.declareParameters(plist, pdlist);
+    assertTrue(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testDeclareParametersBadListLength() {
+    List<ParameterVariant> plist = new ArrayList<ParameterVariant>();
+    List<rcl_interfaces.msg.ParameterDescriptor> pdlist = new ArrayList<rcl_interfaces.msg.ParameterDescriptor>();
+
+    plist.add(new ParameterVariant("foo", true));
+
+    node.declareParameters(plist, pdlist);
+  }
+
+  @Test
+  public final void testUndeclareParameter() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+    node.undeclareParameter("foo");
+    assertFalse(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testUndeclareParameterDoesNotExist() {
+    node.undeclareParameter("foo");
+  }
+
+  @Test
+  public final void testGetParameters() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+    List<ParameterVariant> results = node.getParameters(new ArrayList<String>(Arrays.asList("foo")));
+    assertEquals(1, results.size());
+    assertEquals("foo", results.get(0).getName());
+  }
+
+  @Test
+  public final void testGetParameter() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+    ParameterVariant param = node.getParameter("foo");
+    assertEquals("foo", param.getName());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testGetParameterNotDeclared() {
+    ParameterVariant param = node.getParameter("foo");
+  }
+
+  @Test
+  public final void testGetParameterOr() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    ParameterVariant param = node.getParameterOr("foo", new ParameterVariant("none", false));
+    assertEquals("foo", param.getName());
+  }
+
+  @Test
+  public final void testGetParameterOrNotDeclared() {
+    ParameterVariant param = node.getParameterOr("foo", new ParameterVariant("none", false));
+    assertEquals("none", param.getName());
+  }
+
+  @Test
+  public final void testGetParametersByPrefix() {
+    node.declareParameter(new ParameterVariant("foo.bar", true));
+    node.declareParameter(new ParameterVariant("foo.baz", true));
+    node.declareParameter(new ParameterVariant("doh.baz", true));
+
+    Set<ParameterVariant> params = node.getParametersByPrefix("foo");
+    assertEquals(2, params.size());
```

Comment:
Might be worth asserting that we got the right two.

Reply:
Good call.  In fact, this lead me to completely reexamine getParameterByPrefix and realize that I had it completely wrong.  We really want to return a Map here to be compatible with the rclcpp and rclpy APIs.  Done in 36daa72840b68189f0c6d7d763440c2395ec0d46